### PR TITLE
Add memory limit requests for bazel jobs

### PR DESCRIPTION
**Pull request by krzyzacy**
for https://github.com/kubernetes/test-infra/issues/5449

6Gi for k/k jobs, 2Gi for bazel jobs in other repo

we still need to solve https://github.com/kubernetes/test-infra/issues/5436

/assign @BenTheElder @cjwagner

### CODE REVIEW

File: prow/config.yaml [hunk 1]
```diff
@@ -216,6 +216,9 @@ presubmits:
         ports:
         - containerPort: 9999
           hostPort: 9999
+        resources:
+          requests:
+            memory: "6Gi"
```
Comment (cjwagner) [inline prow/config.yaml]: This is for k/federation not k/k. Did you mean for this to be `2Gi` or is this correct?

**Comment by BenTheElder** [comment]
This seems like a good starting point
/lgtm

File: prow/config.yaml [hunk 2]
```diff
@@ -18187,6 +18280,9 @@ periodics:
       ports:
       - containerPort: 9999
         hostPort: 9999
+      resources:
+        requests:
+          memory: "2Gi"
```
Comment (cjwagner) [inline prow/config.yaml]: Should this be in the `volumes` section? What about`ports`?
The changes below this are similar.

Comment (BenTheElder) [reply prow/config.yaml]: no? the jobs are OOM because they run out of ram. disk is not an issue currently.

Comment (krzyzacy) [reply prow/config.yaml]: https://github.com/krzyzacy/test-infra/blob/42f32bcdcd1a7a12d43fabf3afb1da84b0c4092d/prow/kube/types.go#L133 resources is defined within container

File: prow/config.yaml (hunk 1 above)
Comment (krzyzacy) [reply prow/config.yaml]: `k/fed` is just moved out of `k/k`, to be safe I set that matches rest of the k/k jobs, we can always update that later.

**Comment by cjwagner** [comment]
/hold 

> resources is defined within container

It isn't in every place in this PR though 😄

File: prow/config.yaml (hunk 2 above)
Comment (cjwagner) [reply prow/config.yaml]: Sorry, I wasn't clear. This PR is **currently** adding `resources` and `ports` under `volumes` in some places.

**Comment by krzyzacy** [comment]
@cjwagner we have https://github.com/kubernetes/test-infra/pull/5457 to set default for all other pods to 1Gi already, and bazel jobs need more than that so they need to be explicit

File: prow/config.yaml (hunk 2 above)
Comment (krzyzacy) [reply prow/config.yaml]: grrr, I think the config is correct, but git UI is not expanding properly

File: prow/config/config_test.go [hunk 3]
```diff
@@ -512,6 +512,10 @@ func CheckBazelbuildSpec(t *testing.T, name string, spec *kube.PodSpec, periodic
 				t.Errorf("%s: non-periodic jobs need a --repo=org/$(REPO_NAME) somewhere", name)
 			}
 		}
+
+		if c.Resources.Requests == nil {
```
Comment (krzyzacy) [inline prow/config/config_test.go]: @cjwagner actually this is proving the field exists after parsing the config

File: prow/config.yaml (hunk 2 above)
Comment (BenTheElder) [reply prow/config.yaml]: xref comment below about a new presubmit enforcing this: https://github.com/kubernetes/test-infra/pull/5475#discussion_r150658844

**Pull request merged**